python_modules/libraries/dagster-powerbi/dagster_powerbi/translator.py

```diff
@@ -0,0 +1,86 @@
+from enum import Enum
+from typing import Any, Dict, Generic
+
+from dagster import _check as check
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.asset_spec import AssetSpec
+from dagster._record import record
+from typing_extensions import TypeVar
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+# scaffold out what a generic translator class might look like
+class DagsterTranslator(Generic[T, U]):
+    def __init__(self, context: U):
+        self._context = context
+
+    def get_asset_spec(self, data: T) -> AssetSpec:
+        raise NotImplementedError
```

Comment:
for sake of consistency, im not sure if we should cut over to this pattern right now until other integrations are ready

Reply:
Yeah, depending on feedback in https://github.com/dagster-io/internal/discussions/10935 may make sense to more closely replicate existing translators.

I do think that an experimental integration with limited users is prob the easiest place to make this initial cutover, that way we can get user feedback before cutting over existing integrations.